File: Day 17 05-11-24/solution-1.js
// Define a function using a fat arrow syntax to check for duplicates in an array
const containsDuplicate = (nums) => {
    // Create a new Set to store unique numbers
    const numSet = new Set();
    // Iterate over each number in the input array
    for (let num of nums) {
        // Check if the number is already in the Set
        if (numSet.has(num)) {
            // If it is, return true indicating a duplicate is found
            return true;
        }
        // Otherwise, add the number to the Set
        numSet.add(num);
    }
    // If no duplicates are found, return false
    return false;
}

File: Day 17 05-11-24/LeetCode/solution-1.py
def contains_duplicate(nums):
    """
    Check if there are any duplicates in the list of numbers.

    :param nums: List[int] - A list of integers.
    :return: bool - True if any integer appears at least twice, False otherwise.
    """
    num_set = set()
    for num in nums:
        if num in num_set:
            return True
        num_set.add(num)
    return False
